reference/docs-conceptual/lang-spec/chapter-02.md

```diff
@@ -0,0 +1,1280 @@
+
+Names are partitioned into various namespaces each of which is stored on a virtual drive (§3.1). For
+example, variables are stored on `Variable:`, environment variables are stored on `Env:`, functions
+are stored on `Function:`, and aliases are stored on `Alias:`. All of these names can be accessed as
+variables using the *variable-namespace* production within *variable-scope*. For example,
+
+```powershell
+function F { "Hello from F" }
+$Function:F # invokes function F
+
+Set-Alias A F
+$Alias:A # invokes function F via A
+
+$Count = 10
+$Variable:Count # accesses variable Count
+$Env:Path # accesses environment variable Path
+```
+
+Any use of a variable name with an explicit `Variable:` namespace is equivalent to the use of that
+same variable name without that qualification. For example, `$v` and `$Variable:v` are
+interchangeable.
+
+As well as being defined in the language, variables can also be defined by the cmdlet `New-Variable`
+(§13.37).
+
+#### User-defined variables
+
+Any variable name allowed by the grammar but not used by automatic or preference variables is
+available for user-defined variables.
+
+User-defined variables are created and managed by user-defined script.
+
+#### Automatic variables
+
+Automatic variables store state information about the PowerShell environment. Their values can be
+read in user-written script but not written.
+
+> [!NOTE]
+> The table originally found in this document was removed to reduce duplication. For a complete list
+> of automatic variables, see
+> [about_Automatic_Variables](/powershell/module/microsoft.powershell.core/about/about_automatic_variables).
+
+#### Preference variables
+
+Preference variables store user preferences for the session. They are created and initialized by the
+PowerShell runtime environment. Their values can be read and written in user-written script.
+
+> [!NOTE]
+> The table originally found in this document was removed to reduce duplication. For a complete list
+> of preference variables, see
+> [about_Preference_Variables](/powershell/module/microsoft.powershell.core/about/about_preference_variables).
+
+
+### Commands
+
+Syntax:
+
+```Syntax
+generic-token:
+    generic-token-parts
+
+generic-token-parts:
+    generic-token-part
+    generic-token-parts generic-token-part
+
+generic-token-part:
+    expandable-string-literal
+    verbatim-here-string-literal
+    variable
+    generic-token-char
+
+generic-token-char:
+    *Any Unicode character except
+    { } ( ) ; , | & $
+    ` (The backtick character U+0060)
+    * double-quote-character
+    single-quote-character
+    whitespace
+    new-line-character
+    escaped-character
+
+generic-token-with-subexpr-start:
+    generic-token-parts $(
+```
+
+### Parameters
+
+Syntax:
+
+```Syntax
+command-parameter:
+    dash first-parameter-char parameter-chars
+    colon~opt~ first-parameter-char:
+    A Unicode character of classes Lu, Ll, Lt, Lm, or Lo
+    _ (The underscore character U+005F)
+    ?
```

Comment:
Should `colon~opt~` be part of the previous line, and `first-parameter-char:` aligned to the start of the line?

Reply:
Yes, thanks for spotting that. Conversion from the Word doc required a lot of cleanup. I am sure I missed many more. More edits to com.